### PR TITLE
show notification about tx inflight and action required

### PR DESCRIPTION
**Pull request by troggy**
image

### CODE REVIEW

File: src/components/txNotification/index.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { TxStatus } from './types';
+
+
+const statusDetails = {};
+statusDetails[TxStatus.CREATED] = { text: 'Waiting for signature..', icon: 'warning' };
+statusDetails[TxStatus.INFLIGHT] = { text: 'Mining..', icon: 'info' };
+statusDetails[TxStatus.SUCCEED] = { text: 'Success', icon: 'success' };
+statusDetails[TxStatus.FAILED] = { text: 'Transaction failed', icon: 'error' };
+statusDetails[TxStatus.CANCELLED] = { text: 'Cancelled', icon: 'error' };
+
```
Comment (sunify) [inline src/components/txNotification/index.tsx]:
```es6
const statusDetails = {
  [TxStatus.CREATED]: { text: 'Waiting for signature..', icon: 'warning' },
  [TxStatus.INFLIGHT]: { text: 'Mining..', icon: 'info' },
  [TxStatus.SUCCEED]: { text: 'Success', icon: 'success' },
  [TxStatus.FAILED]: { text: 'Transaction failed', icon: 'error' },
  [TxStatus.CANCELLED]: { text: 'Cancelled', icon: 'error' },
};
```

:-)

Comment (troggy) [reply src/components/txNotification/index.tsx]: Cool! Thanks. What kind of syntax is this?

Comment (sunify) [reply src/components/txNotification/index.tsx]: Computed property keys from ES6. You can put any expression in square brackets:

```
console.log({ [Math.random()]: 'random key' });
```

File: src/stores/token.ts [hunk 2]
```diff
@@ -33,18 +34,20 @@ export default class Token extends ContractStore {
...
 
   public color: number;
   @observable public name: string;
   @observable public symbol: string;
   @observable public decimals: number;
   @observable public balance?: number;
 
-  constructor(account: Account, address: string, color: number) {
+  constructor(account: Account, txs: Transactions, address: string, color: number) {
     super(tokenAbi, address);
```
Comment (sunify) [inline src/stores/token.ts]: You need to pass `txs` as a 3rd argument to `super` (and not assign `this.txs` at 50)

File: src/stores/bridge.ts [hunk 3]
```diff
@@ -66,8 +67,8 @@ export default class Bridge extends ContractStore {
   @observable public slots: IObservableArray<Slot> = observable.array([]);
   @observable public lastCompleteEpoch: number;
 
-  constructor(account: Account, address?: string) {
-    super(bridgeAbi, address);
+  constructor(account: Account, transactions: Transactions, address?: string) {
```
Comment (sunify) [inline src/stores/bridge.ts]: Here `transactions` in other places `txs`. Let's use one of that, not both

File: src/components/txNotification/transactions.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
+/**
```
Comment (sunify) [inline src/components/txNotification/transactions.ts]: Why this file not in `stores` folder?

Comment (troggy) [reply src/components/txNotification/transactions.ts]: I thought to make this component+store reusable and ready to drop in other projects. What do you think? Is it a good idea? What are the best practices for this?

Comment (troggy) [reply src/components/txNotification/transactions.ts]: also that's why I created a separate type definition `DetailedInflightTxReceipt` so this "component" doesn't depend on types used in host project (`InflightTxReceipt`).

Yes, these names suck. I tried to make the whole thing simpler to understand and to use, but this "promise of promise of tx receipt" mess still gives me some headache

File: src/stores/bridge.ts (hunk 3 above)
Comment (troggy) [reply src/stores/bridge.ts]: good catch!

File: src/stores/token.ts (hunk 2 above)
Comment (troggy) [reply src/stores/token.ts]: 🙏

File: src/components/txNotification/transactions.ts (hunk 4 above)
Comment (sunify) [reply src/components/txNotification/transactions.ts]: Ok, makes sense

**Pull request merged**